### ISSUE TITLE
Using compatible shim and go-apex releases

### ISSUE THREAD

**Issue by myles-mcdonnell** [apex/go-apex#65]
TL;DR
How can it be programatically determined which releases of go-apex are compatible with which versions of https://github.com/apex/apex?

========

Is there a preferred technique for ensuring that compatible versions of https://github.com/apex/go-apex and https://github.com/apex/apex are used together?

Our pipeline automates infrastructure and app. deployment.  Currently we use https://raw.githubusercontent.com/apex/apex/707e92374bbaf1de07bc29c0272b047f3972b45f/install.sh during CI which pulls master.  We use https://github.com/golang/dep in our app repo to define the version of go-apex we want.  

We could I suppose configure the dep to always pull the latest go-apex and hope that the latest of both are always compatible, however we would prefer to lock in compatible versions of both and upgrade explicitly as required.

**Comment by pasztorpisti** [comment]
In my opinion the correct solution would be vendoring our tools (not only apex) in some way.

One of the go projects I worked on had a github repo that contained versioned git submodules (pinning the correct version of tool sources) and a script that installed the tools for the developer either by compiling the pinned sources or by downloading the correct release binary.

With that solution things could still go wrong (not updating the installed tools on a dev machine for a long time) but there was at least a quick fix (reinstall) and a single source of truth when something went wrong.
